File: FMI-Materials/src/main/java/com/fmi/materials/model/Section.java
package com.fmi.materials.model;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

@Getter
@Setter
@EqualsAndHashCode
@ToString
@Entity
@Table(name = "sections")
public class Section {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    private Course course;

    @OneToMany(mappedBy = "section", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Material> materials;

    public Section() {}

    public Section(String name, Course course, List<Material> materials) {
        this(null, name, course, materials);
    }

    public Section(Long id, String name, Course course, List<Material> materials) {
        this.id = id;
        this.name = name;
        this.course = course;
        this.materials =  (materials != null ? materials.stream().collect(Collectors.toSet()) : null);
    }
}

File: mailer/src/model/Section.cs
using System;
using System.Collections.Generic;

namespace mailer;

public partial class Section
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public int CourseId { get; set; }

    public virtual Course Course { get; set; } = null!;

    public virtual ICollection<MaterialRequest> MaterialRequests { get; } = new List<MaterialRequest>();

    public virtual ICollection<Material> Materials { get; } = new List<Material>();
}
